transitions/registration.js

```diff
@@ -217,21 +216,29 @@ module.exports = {
                 if (self.isBoolExprFalse(obj, event.bool_expr)) {
                     return;
                 }
-                var options = { db: db, audit: audit, doc: doc };
-                if (event.params) {
-                    // params setting get sent as array
+                var options = { db: db, audit: audit, doc: doc, registrationConfig: registrationConfig };
+
+                if (typeof event.params === 'string') {
+                    // params setting can get sent as string to convert into an array
                     options.params = event.params.split(',');
+                } else if (typeof event.params === 'object') {
```

Comment:
That's what real testing is for! :P 

But right OK, that's awful. I might need to re-think this. Do you know how this stuff works? Can I make the type both?